File: WorldFeed/Server/src/WorldFeed.Common/Public/Models/Interfaces/DTO/ITwitterListDTO.cs
namespace WorldFeed.Common.Public.Models.Interfaces.DTO
{
    using System;

    using WorldFeed.Common.Public.Models.Enums;

    public interface ITwitterListDTO : ITwitterListIdentifier
    {
        string IdStr { get; set; }

        string Name { get; set; }
        string FullName { get; set; }

        IUserDTO Owner { get; set; }
        DateTime CreatedAt { get; set; }
        string Uri { get; set; }
        string Description { get; set; }
        bool Following { get; set; }
        PrivacyMode PrivacyMode { get; set; }

        int MemberCount { get; set; }        
        int SubscriberCount { get; set; }
    }
}

File: Sharebook/Client/src/app/core/Public/Models/Interfaces/DTO/ITwitterListDTO.ts
import {InjectionToken} from "@angular/core";

import {ITwitterListIdentifier} from "../ITwitterListIdentifier";
import {IUserDTO} from "./IUserDTO";
import { PrivacyMode } from '../../Enum/PrivacyMode';
import {TwitterListDTO} from "../../../../Core/DTO/TwitterListDTO";
import DateTime from "typescript-dotnet-commonjs/System/Time/DateTime";

export interface ITwitterListDTO extends ITwitterListIdentifier {
  idStr: string;

  name: string;
  fullName: string;

  owner: IUserDTO;
  createdAt: DateTime; // DateTimeOffset;
  uri: string;
  description: string;
  following: boolean;
  privacyMode: PrivacyMode;

  memberCount: number;
  subscriberCount: number;
}

export const ITwitterListDTOToken = new InjectionToken<ITwitterListDTO>('ITwitterListDTO', {
  providedIn: 'root',
  factory: () => new TwitterListDTO(),
});
